src/components/Group/GroupMemberCard.tsx

```diff
@@ -0,0 +1,146 @@
+import { Role } from "@prisma/client";
+import { PublicUser } from "../../utils/types";
+import { useContext } from "react";
+import { UserContext } from "../../utils/userContext";
+import Spinner from "../Spinner";
+import _ from "lodash";
+import { trpc } from "../../utils/trpc";
+import { toast } from "react-toastify";
+import { TRPCClientError } from "@trpc/client";
+import { useToasts } from "react-toast-notifications";
+
+interface GroupMembersProps {
+  users: PublicUser[];
+  onClose: () => void;
+}
+export const GroupMembers = (props: GroupMembersProps) => {
+  const curUser = useContext(UserContext);
+  const driver = props.users.find((user) => user.role === Role.DRIVER);
+  const riders = props.users.filter(
+    (user) => user.id !== driver?.id && user.id !== curUser?.id
+  );
+  const utils = trpc.useContext();
+  const { addToast } = useToasts();
+
+  const { mutate: deleteGroup } = trpc.user.groups.delete.useMutation({
+    onError: (error: any) => {
+      toast.error(`Something went wrong: ${error.message}`);
+    },
+    onSuccess() {
+      utils.user.groups.me.invalidate();
+      utils.user.me.invalidate();
+      props.onClose();
+      addToast("Group has been successfully deleted");
+    },
+  });
+
+  const { mutate: editGroup } = trpc.user.groups.edit.useMutation({
+    onError: (error: any) => {
+      toast.error(`Something went wrong: ${error.message}`);
+    },
+    onSuccess() {
+      utils.user.groups.me.invalidate();
+      utils.user.me.invalidate();
+      props.onClose();
+      addToast("Removed from group");
+    },
+  });
+
+  if (!driver || !curUser) {
+    return <Spinner />;
+  }
+
+  const handleDelete = (id: string) => {
+    if (driver.carpoolId) {
```

Comment:
How frequently do you expect variables in this component to be updated? I'm not as familiar with some of the new libraries, but if there are a lot of re-renders of this then it might be useful to use the useCallback hook.